### DISCUSSION TITLE
npm i -D @sveltejs/kit@next Errors

### DISCUSSION THREAD

**Discussion by MrAmericanMike** [sveltejs/kit#6584] [General]
Sorry if this is not a problem with SvelteKit itself and maybe it's NPM fault. But I'm surprised to see this errors while installing a package.

Why it errors with complains that should show, if any, when running `vite dev` for example.

```
C:\VSCode\Svelte\kit-svelte  
λ npm i -D @sveltejs/kit@next
npm ERR! code 1
npm ERR! path C:\VSCode\Svelte\kit-svelte\node_modules\@sveltejs\kit
npm ERR! command failed
npm ERR! command C:\Windows\system32\cmd.exe /d /s /c node svelte-kit.js sync
npm ERR! > config.kit.files.template has been renamed to config.kit.files.appTemplate
npm ERR!     at file:///C:/VSCode/Svelte/kit-svelte/node_modules/@sveltejs/kit/src/core/config/options.js:497:10
npm ERR!     at file:///C:/VSCode/Svelte/kit-svelte/node_modules/@sveltejs/kit/src/core/config/options.js:389:43
npm ERR!     at file:///C:/VSCode/Svelte/kit-svelte/node_modules/@sveltejs/kit/src/core/config/options.js:375:18
npm ERR!     at file:///C:/VSCode/Svelte/kit-svelte/node_modules/@sveltejs/kit/src/core/config/options.js:375:18
npm ERR!     at file:///C:/VSCode/Svelte/kit-svelte/node_modules/@sveltejs/kit/src/core/config/options.js:375:18
```

It also complained about `config.kit.browser.hydrate has been removed.` 

Is this behavior normal?

**Comment by infovore** [comment]
No, this is the current behaviour: it is currently set up such that it won't install if it encounters some invalid configurations (for instance, this error in `sveltekit.config.js`. it similarly errors when trying to be installed to 'old-style' routing, which must be migrated before > `next.405` versions will install. I'm not in a place to pass judgment on why this has been chosen.
